pf_driver/src/pf/pfsdp_protocol.cpp

```diff
@@ -204,6 +203,12 @@ void PFSDPBase::declare_critical_parameters()
   descriptorPort.name = "Initial IP port";
   descriptorPort.description = "See address";
   node_->declare_parameter<int>("port", 0, descriptorPort);
+
+  rcl_interfaces::msg::ParameterDescriptor descriptorTransport;
+  descriptorTransport.name = "IP transport port";
+  descriptorTransport.description = "Can be either tcp or udp";
+  descriptorTransport.read_only = true;
+  node_->declare_parameter<std::string>("transport", "tcp", descriptorTransport);
```

Comment:
Also I didn't notice this earlier, but it's better to separate the protocol-specific code from ROS-specific code. The protocol-specific code is intended to be used like a pure cpp library with no ROS dependency, apart from logging.